src/app/qgisapp.cpp

```diff
@@ -4973,6 +4973,11 @@ void QgisApp::initLayerTreeView()
 
   connect( mMapCanvas, &QgsMapCanvas::mapCanvasRefreshed, this, &QgisApp::updateFilterLegend );
   connect( mMapCanvas, &QgsMapCanvas::renderErrorOccurred, badLayerIndicatorProvider, &QgsLayerTreeViewBadLayerIndicatorProvider::reportLayerError );
+  connect( mMapCanvas, &QgsMapCanvas::renderErrorOccurred, mInfoBar, [this]( const QString & error, QgsMapLayer * layer )
+  {
+    QString message = layer->name() + QStringLiteral( ": " ) + error;
+    mInfoBar->pushItem( new QgsMessageBarItem( message, Qgis::MessageLevel::Warning, 60 ) );
```

Comment:
```suggestion
    mInfoBar->pushWarning( layer->name(), error );
```

For a couple of reasons:
1. There should be no timeout for warning/error messages, they must always be manually dismissed
2. ':' isn't universal -- it needs translating too. But here it's better to just use the "title" of the message as the layer name so it gets formatted nicely